ert_shared/storage/extraction.py

```diff
@@ -350,9 +354,13 @@ def post_ensemble_results(ensemble_id: str) -> None:
 
 
 @feature_enabled("new-storage")
-def post_ensemble_data(ensemble_size: int, update_id: str = None) -> str:
+def post_ensemble_data(
```

Comment:
Could you explain the motivation for keeping it optional? Could we not have the caller pass the global object if that is to be used?

Reply:
It was optional before by not existing. What you're suggesting would require us to change every caller of `post_ensemble_data` and thus make the PR bigger. I can do that of course, but you'll need to give me a thumbs up.

Reply:
Yeah, but that would unify the usage in tests and code. I think that would be better 👍🏻 But keeping that change in a separate commit, at least during review, is probably helpful...

Reply:
I agree to this and was about to make the same comment, but as @dotfloat points out, it will make this PR bigger. But from my point of view i would say it is worth it! I very much like dependency injection instead of the function getting things itself 👍

Reply:
Sure it will make the PR bigger, but it will also remove the knowledge of the global ert instance from the exporter and make the code path for the tests the same as used in the rest of the codebase... Go for it 🚀

Reply:
Okie